### ISSUE TITLE
On possibility of updating to RDF4J

### ISSUE THREAD

**Issue by schivmeister** [Complexible/pinto#20]
I was wondering why, instead of updating to RDF4J, Pinto was only updated to Sesame 4.x? Was this a conscious decision based on certain things or just a preference?

http://docs.rdf4j.org/migration/

**Comment by mhgrove** [comment]
just a preference, mostly because that's the version we've standardized on

**Comment by schivmeister** [comment]
I don't think it's that much work from 4.x, but I won't know until I dig. It looks like namespace changes mostly, but there could be other regressions waiting. But you have a nice test suite already that should cover that.

By any chance, if I chose to pursue an update myself, I won't be able to do it since your Sesame dependencies are coming from Stardog repositories, correct?

**Comment by mhgrove** [comment]
no, this is not used in stardog, so it could be upgraded without affecting anything else

**Comment by schivmeister** [comment]
@mhgrove So, I finally got to this, but I am not sure if https://github.com/mhgrove/cp-openrdf-utils is up-to-date or there are some internal changes that you didn't make public. The latest version there is `4.0.1` but here it's `4.0.2`.

After updating the namespaces to RDF4J, the build and all but two tests pass in there. And here, build fails because `Models2.toList()` is having an incompatible signature `(List<Value>, Model)` vs. `(List<Resource>)`. `UUIDCodec.writeValue()` also has an incompatible return type `ResourceBuilder` vs. `Value`.

There appear to be no other errors (there was one wrt `URI`, but I changed that to `IRI` and it became happy). So I think it's not too much work to update to RDF4J as this looks close. I can send a pull request for both projects if you can help clarify those remaining issues.

**Comment by mhgrove** [comment]
not sure what the version mismatch is for the openrdf utils, but probably not a big deal. the migration sounds good otherwise

**Comment by schivmeister** [comment]
https://github.com/mhgrove/cp-openrdf-utils/pull/23
